modules/engines/engine-javascript-graalium/src/main/resources/polyfills/require.js

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2022 SAP SE or an SAP affiliate company and Eclipse Dirigible contributors
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-FileCopyrightText: 2022 SAP SE or an SAP affiliate company and Eclipse Dirigible contributors
+ * SPDX-License-Identifier: EPL-2.0
+ */
+var Require = (function (modulePath) {
+    var _loadedModules = {};
+    var _require = function (path) {
+        var moduleInfo, buffered, head = '(function(exports,module,require){ ',
+            code = '',
+            tail = '})',
+            line = null;
+        moduleInfo = _loadedModules[path];
+        if (moduleInfo) {
+            return moduleInfo;
+        }
+        code = Java.type('org.eclipse.dirigible.engine.js.graalvm.execution.js.modules.DirigibleModuleProvider').loadSource(path);
+        moduleInfo = {
+            loaded: false,
+            id: path,
```

Comment:
Could this be used to set the fileName for the exception handling?
- https://github.com/SAP/xsk/issues/1426
- https://github.com/SAP/xsk/pull/1427

Reply:
In some cases, yes but I'm still working on an implementation of a `require` that is implemented in Java and uses some Graal APIs that would solve the issues related to no source metadata persisted (like the one you mentioned) and the code modifications we do that negatively impact the debugging.